docs/PythonAPI.md

```diff
@@ -0,0 +1,867 @@
+
+## Class `Runtime` 
+
+The `Runtime` class coordinates the devices with the storage disc including selecting the devices, setting their properties, and starting and stopping acqusition.
+
+```python
+class Runtime:
+    def __init__(self, *args: None, **kwargs: Any) -> None:
+        """Initializes the Runtime object with optional arguments."""
+    
+    def device_manager(self) -> DeviceManager:
+        """Returns the DeviceManager instance associated with this Runtime."""
+    
+    def get_available_data(self, stream_id: int) -> AvailableData:
+        """Returns the AvailableData instance for the given stream ID.
+        
+        Args:
+            stream_id (int): The ID of the stream for which available data is requested.
+            
+        Returns:
+            AvailableData: The AvailableData instance for the given stream ID.
+        """
+    
+    def get_configuration(self) -> Properties:
+        """Returns the current configuration properties of the runtime."""
+    
+    def get_state(self) -> DeviceState:
+        """Returns the current state of the device."""
+    
+    def set_configuration(self, properties: Properties) -> Properties:
+        """Applies the provided configuration properties to the runtime.
+        
+        Args:
+            properties (Properties): The properties to be set.
+            
+        Returns:
+            Properties: The updated configuration properties.
+        """
+    
+    def start(self) -> None:
+        """Starts the runtime, allowing it to collect data."""
+    
+    def stop(self) -> None:
+        """Stops the runtime, ending data collection after the max number of frames is collected."""
+    
+    def abort(self) -> None:
+        """Aborts the runtime, terminating it immediately."""
+```
+
+- Call `device_manager()` to return the `DeviceManager` object associated with this `Runtime` instance. 
+
+- Call `get_available_data` with a specific `stream_id` to return the `AvailableData` associated with the `stream_id`.
+
+- Call `get_configuration()` to return the `Properties` object associated with this `Runtime` instance.
+
+- Call `get_state()` to return the `DeviceState` object associated with this `Runtime` instance.
+
+- Call `set_configuration` with a `Properties` object to change the properties of this `Runtime` instance.
+
+- Call `start()` to begin data acquisition.
+
+- Call `stop()` to end data acquisition once the max number of frames specified in `config.video[0].max_frame_count` is collected.
+
+- Call `abort()` to  `Runtime` instance.
+
+Note: If you set the maximum frame count to 0, `runtime.stop()` will hang forever. In order to terminate acquisition in this case, you need to call `runtime.abort()` instead, which will stop acquisition immediately. You can also use `runtime.abort()` if you set the maximum frame count to some nonzero value but you want to terminate acquisition before you reach that maximum frame count.
```

Comment:
This is my mistake. While I believe this was true at one point, it's no longer true.